src/core/components/SearchHistory.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState, useRef, useEffect } from 'react';
+import styles from './SearchChatHistory.module.css';
+import ModalWrapper from './ModalWrapper';
+import SearchHistoryModalBody from './SearchHistoryModalBody';
+import { useAppContext } from '../context/useAppContext';
+import { SearchHistoryButton } from '../assets/SearchHistoryButton';
+import { ConversationHistory } from '../context/types';
+
+interface SearchHistoryProps {
+  isSearchHistoryOpen: boolean;
+  setIsSearchHistoryOpen: (i: boolean) => void;
+  setIsMobileSideBarOpen: (i: boolean) => void;
+}
+
+const SearchHistory = ({
```

Comment:
This used to be `SearchChatHistory` but I dropped the redundant '-chat-' in the name.